src/NCRings.jl

```diff
@@ -83,15 +83,22 @@ end
 
 *(x::NCRingElement, y::NCRingElem) = parent(y)(x)*y
 
-function ==(x::NCRingElem, y::NCRingElem)
-   fl, u, v = try_promote(x, y)
```

Comment:
Let's look into retaining this, and documenting this promotion system (and then perhaps we can also use it to simplify some other code which does ad-hoc stuff right now)